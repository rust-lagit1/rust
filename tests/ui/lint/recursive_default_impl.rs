//@check-fail
struct Foo{
    bar: u32,
}

impl Default for Foo {
    fn default() -> Self {
    //~^ ERROR: recursive default impl
    //~| NOTE: will result in infinite recursion
    //~| HELP: ..default() in the Default impl does not apply a default for each struct field
    //~| NOTE: `#[deny(recursive_default_impl)]` on by default
        Self {
            ..Default::default()
            //~^ NOTE: recursive call site

        }
    }
}

fn main() {
    let _ = Foo::default();
}
