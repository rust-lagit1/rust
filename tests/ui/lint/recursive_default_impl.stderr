error: recursive default impl
  --> $DIR/recursive_default_impl.rs:7:5
   |
LL |     fn default() -> Self {
   |     ^^^^^^^^^^^^^^^^^^^^ will result in infinite recursion
...
LL |             ..Default::default()
   |               ------------------ recursive call site
   |
   = help: ..default() in the Default impl does not apply a default for each struct field
   = note: `#[deny(recursive_default_impl)]` on by default

error: aborting due to 1 previous error

